compute_tools/src/lsn_lease.rs

```diff
@@ -0,0 +1,187 @@
+use anyhow::bail;
+use anyhow::Context;
+use anyhow::Result;
+use postgres::{NoTls, SimpleQueryMessage};
+use std::time::SystemTime;
+use std::{str::FromStr, sync::Arc, thread, time::Duration};
+use utils::id::TenantId;
+use utils::id::TimelineId;
+
+use compute_api::spec::ComputeMode;
+use tracing::{info, warn};
+use utils::{
+    lsn::Lsn,
+    shard::{ShardCount, ShardNumber, TenantShardId},
+};
+
+use crate::compute::{ComputeNode, ComputeState};
+
+/// Spawns a background thread to periodically renew LSN leases for static compute.
+/// Do nothing if the compute is not in static mode.
+pub fn launch_lsn_lease_bg_task_for_static(compute: &Arc<ComputeNode>) {
+    let lsn = {
+        let state = compute.state.lock().unwrap();
+        let spec = state.pspec.as_ref().expect("spec must be set");
+        match spec.spec.mode {
+            ComputeMode::Static(lsn) => lsn,
+            _ => return,
+        }
+    };
+    let compute = compute.clone();
+    thread::spawn(move || {
+        if let Err(e) = lsn_lease_bg_task(compute, lsn) {
+            // TODO: might need stronger error feedback than logging an warning.
+            warn!("lsn_lease_bg_task failed: {e}");
```

Comment:
So initially like this, but surely we should kill the compute eventually?

Reply:
Posted some similar thoughts bellow